src/main/java/frc/robot/utils/ControllerJSONReader.java

```diff
@@ -0,0 +1,274 @@
+        HashMap<String, DoubleSupplier> axisList = new HashMap<String, DoubleSupplier>();
+        Iterator<JSONObject> iterator = axisJSON.iterator();
+        while (iterator.hasNext()) {
+            JSONObject axis = iterator.next();
+            DoubleSupplier t;
+
+            int port = ((Long) axis.get("controller")).intValue();
+
+            // first case is for xbox controller, second case is for joysticks
+            switch (((String) axis.get("axis"))) {
+                case "leftX":
+                case "xAxis":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(0);
+                    else t = () -> controllers.get(port).getRawAxis(0);
+                    SmartDashboard.putNumber("readaxis", t.getAsDouble());
+                    break;
+                case "leftY":
+                case "yAxis":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(1);
+                    else t = () -> controllers.get(port).getRawAxis(1);
+                    break;
+                case "leftTrigger":
+                case "zRotate":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(2);
+                    else t = () -> controllers.get(port).getRawAxis(2);
+                    break;
+                case "rightTrigger":
+                case "slider":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(3);
+                    else t = () -> controllers.get(port).getRawAxis(3);
+                    break;
+                case "rightX":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(4);
+                    else t = () -> controllers.get(port).getRawAxis(4);
+                    break;
+                case "rightY":
+                    if ((boolean) axis.get("negate"))
+                        t = () -> -controllers.get(port).getRawAxis(5);
+                    else t = () -> controllers.get(port).getRawAxis(5);
+                    break;
+                default:
+                    t = null;
+                    break;
+            }
+
+            axisList.put((String) axis.get("command"), t);
+        }
+        axes = axisList;
+        return axisList;
+    }
+
+    // assigns pov intsuppliers to string keys
+    private static HashMap<String, IntSupplier> setPOV(JSONArray povJSON) {
+        HashMap<String, IntSupplier> povList = new HashMap<String, IntSupplier>();
+        Iterator<JSONObject> iterator = povJSON.iterator();
+        while (iterator.hasNext()) {
+            JSONObject p = iterator.next();
+
+            if (controllers.get(p.get("controller")) == null) {
+                povList.put((String) p.get("command"), () -> 0);
+            } else
```

Comment:
...and add a warning here.

Reply:
Okay! I'll fix this as soon as possible! Is the robot ready yet?

Reply:
We should get it on Saturday